controllers/complianceeventsapi/server.go

```diff
@@ -446,9 +503,145 @@ func parseQueryArgs(queryArgs url.Values, isCSV bool) (*queryOptions, error) {
 		}
 	}
 
+	authorizedClusterIds, authorizedClusterNames,
+		unAuthorizedClusters, includeAllManagedClusters, err := getAuthorizedClusters(ctx,
```

Comment:
Rather than return all these values, I think it'd be cleaner if `getAuthorizedClusters` directly set the filter values for the clusters and returned an error that included the unauthorized clusters in the text which could be directly used by the calling function of `parseQueryArgs`.